### ISSUE TITLE
Issues with Svelte 4

### ISSUE THREAD

**Issue by punyflash** [inertiajs/inertia#1621]
### Version:

- `@inertiajs/svelte` version: 1.0.9
- `svelte` version: 4

### Describe the problem:

When using Svelte 4, props are not being populated on inertia router visits. Also when adding svelte-eslint to the project pages are completely empty after visit.

### Steps to reproduce:

Just make a visit using inertia link or router to the route that requires any props

**Comment by reinink** [comment]
Hey! I am not able to reproduce this issue. We actually have a Svelte playground included in the project, which is on Svelte 4.x, and I just tested it and it works fine:

https://github.com/inertiajs/inertia/tree/master/playgrounds/svelte

I am going to close this issue for now, but feel free to provide us with a minimal reproduction of the issue you're experiencing as a Git repo and we can have another look 👍

**Comment by punyflash** [comment]
@reinink here is an example repository: https://github.com/punyflash/inertia-test.

## A bit detailed explanation
I created 2 pages: `/page/1` and `page/2` to show how it works. It seems like page is not being updated after persistent layout change

### Page 1

```svelte
<script lang="ts" context="module">
    // import LayoutInsideTag from "@/components/LayoutInsideTag.svelte";
    import LayoutOutsideTag from "@/components/LayoutOutsideTag.svelte";

    export const layout = [LayoutOutsideTag];
</script>

<script lang="ts">
    import { fade } from 'svelte/transition';
    import { inertia } from "@inertiajs/svelte";
    export let title = '';
    export let data = '';
</script>

<svelte:head>
    <title>{title}</title>
</svelte:head>

<section in:fade>
    <h1 class="text-3xl">{title}</h1>
    <div>{data}</div>
    <a href="/page/2" class="underline" use:inertia>Go to page 2</a>
</section>
```

### Page 2

```svelte
<script lang="ts" context="module">
    // import LayoutOutsideTag from "@/components/LayoutOutsideTag.svelte";
    import LayoutInsideTag from "@/components/LayoutInsideTag.svelte";

    export const layout = [LayoutInsideTag];
</script>

<script lang="ts">
    import { fade } from 'svelte/transition';
    import { inertia } from "@inertiajs/svelte";
    export let title = '';
    export let data = '';
</script>

<svelte:head>
    <title>{title}</title>
</svelte:head>

<section in:fade>
    <h1 class="text-3xl">{title}</h1>
    <div>{data}</div>
    <a href="/page/1" class="underline" use:inertia>Go to page 1</a>
</section>
```

## Expected behavior (Svelte 3)
![7AIMHyKTie](https://github.com/inertiajs/inertia/assets/14828695/7d58961b-b5da-445b-ab4d-50e1a5b3a841)

## Actual (Svelte 4)
![ufp6GbXL7f](https://github.com/inertiajs/inertia/assets/14828695/2c64e346-4e4c-47cb-a22a-38d671c31505)

**Comment by bbauti** [comment]
Hey! any updates on this?

**Comment by pedroborges** [comment]
Hey @punyflash, thanks for reporting this bug. At my company we were upgrading to Svelte 4 and ran into the same issue with persistent layouts. My coworker found your issue and confirmed it using the Svelte playground.

I was able to identify that this issue started happening in Svelte [v4.1.1](https://github.com/sveltejs/svelte/releases/tag/svelte%404.1.1). The solution for now is to force Svelte v4.1.0.

I won't have time to work on a fix in the next two weeks. When I get a chance I will try to determine if this is a bug on our side or Svelte's.

On our side it could be related to the [key part of this each block](https://github.com/inertiajs/inertia/blob/a650a3771aa0ff64da0187030fa47d24084dd957/packages/svelte/src/Render.svelte#L21) 🤷‍♂️ 

cc/ @bbauti

**Comment by rkyoku** [comment]
Hi there,

A nice fellow on Discord sent me here, kudos to him/her!

I can confirm the bug with both versions 4.1.1 and 4.1.0 of Svelte, although the behavior is different.

On version 4.1.1 the main `<slot/>` never gets updated after a redirect (login/logout), although the layout is updated (and so are the props used in the layout, e.g. my `<h1>` tag). I have to refresh (F5) the page for the main slot to be updated again.

On version 4.1.0 it works better, but still fails if I do the login from a page other than `/`, in which case the props are not passed to the template, and I get errors like "#each should only be used on iterables" things like that.

Downgrading to Svelte 4.0.0 works properly for me! I'll stick to version 4.0.0 until things are fixed on either side.

Good luck with this bug, and thanks for this great tool!

Best,

**Comment by Josefikos** [comment]
svelte v5 won't work too

**Comment by delordemm1** [comment]
> svelte v5 won't work too

Damn, i thought it was from my end, battled this for hours 🥲

**Comment by punyflash** [comment]
As it seems the package is dead and most of crucial PRs are hanging for almost a year (#1614, #1763, #1635), I created my own adapter for Svelte with typescript support and some issues fixed regarding styles and props loading. You are free to use it: https://github.com/westacks/inertia-svelte. You can natively replace the inertia adapter, and it will work as it is.

**Comment by vitalijalbu** [comment]
Coll, what makes the maintainer of this library not update it anymore? At least can the public make pr?

**Comment by Dibbyo456** [comment]
> You are free to use it: https://github.com/westacks/inertia-svelte. You can natively replace the inertia adapter, and it will work as it is.

@punyflash How do I install it?

**Comment by punyflash** [comment]
> @punyflash How do I install it?

```sh
npm i @westacks/inertia-svelte
```

**Comment by Dibbyo456** [comment]
I am getting errors (not using TS).

```
    ~/Pr/inertia-bug-report    main !4  npm run dev                                                         1 ✘  7s  

> dev
> vite


  VITE v5.0.12  ready in 217 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
✘ [ERROR] /Users/harry/Projects/inertia-bug-report/node_modules/@westacks/inertia-svelte/dist/components/Link.svelte:2:14 Unexpected token [plugin vite-plugin-svelte:optimize-svelte]

    node_modules/@westacks/inertia-svelte/dist/components/Link.svelte:2:14:
      2 │  import type { Method, PreserveStateOption, RequestPayload } from '@inertiajs/core'
        ╵               ^

  The plugin "vite-plugin-svelte:optimize-svelte" was triggered by this import

    node_modules/@westacks/inertia-svelte/dist/index.js:1:67:
      1 │ ..."@inertiajs/core";import{default as Te}from"./components/Link.svelte";import{router as v,setupProgress as P}from...
        ╵                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] /Users/harry/Projects/inertia-bug-report/node_modules/@westacks/inertia-svelte/dist/components/App.svelte:2:14 Unexpected token [plugin vite-plugin-svelte:optimize-svelte]

    node_modules/@westacks/inertia-svelte/dist/components/App.svelte:2:14:
      2 │  import type { Page } from '@inertiajs/core'
        ╵               ^

  The plugin "vite-plugin-svelte:optimize-svelte" was triggered by this import

    node_modules/@westacks/inertia-svelte/dist/index.js:1:167:
      1 │ ...ess as P}from"@inertiajs/core";import E from"./components/App.svelte";import k from"./components/SSR.svelte";imp...
        ╵                                                ~~~~~~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] /Users/harry/Projects/inertia-bug-report/node_modules/@westacks/inertia-svelte/dist/components/SSR.svelte:2:14 Unexpected token [plugin vite-plugin-svelte:optimize-svelte]

    node_modules/@westacks/inertia-svelte/dist/components/SSR.svelte:2:14:
      2 │  import type { Page } from '@inertiajs/core'
        ╵               ^

  The plugin "vite-plugin-svelte:optimize-svelte" was triggered by this import

    node_modules/@westacks/inertia-svelte/dist/index.js:1:206:
      1 │ ... from"./components/App.svelte";import k from"./components/SSR.svelte";import F from"./store";async function S({i...
        ╵                                                ~~~~~~~~~~~~~~~~~~~~~~~~~


  LARAVEL v10.43.0  plugin v1.0.1

  ➜  APP_URL: http://localhost
```

**Comment by punyflash** [comment]
> I am getting errors (not using TS).

Update package to 1.0.20, it should work now even if you don't use typescript

**Comment by Dibbyo456** [comment]
Thank you, it works now. 👍

If you don't mind, can you take a look at issue #1785

**Comment by joshstobbs** [comment]
@Dibbyo456 I put in a PR over at the westacks repo that solves the unknown prop issue.  Still trying to sort through the unexpected slot part but I'm not too sure where that's coming from, will keep this updated if I can figure it out.

**Comment by Dibbyo456** [comment]
@joshstobbs thanks for the PR. FYI, when using Svelte 5 the warning do not appear.

**Comment by joshstobbs** [comment]
Oh that’s really cool @Dibbyo456!  I haven’t actually been able to get Inertia to work with Svelte 5, any chance you have an example repo?

**Comment by Dibbyo456** [comment]
https://danmatthews.me/posts/svelte-v5-alpha-inertia

**Comment by driesvints** [comment]
Seems a PR was sent in for this: https://github.com/inertiajs/inertia/pull/1763. Let's see how it goes. Thanks all.

**Comment by reinink** [comment]
Hey folks! Just merged in #1763 — let's hope this is all resolved now. Will try and get a release out shortly 👍

**Comment by nairboon** [comment]
> Just follow [this guide](https://danmatthews.me/posts/svelte-v5-alpha-inertia), unfortunately SSR is broken #1801 😕

This guide from @danmatthews seems to have moved to this url: https://danmatthews.me/posts/you-can-use-svelte-v5-with-inertiajs-really-easily-today-bf9de
